src/my-bot.ts

```diff
@@ -1,38 +1,423 @@
+                return place
+            }
+        } while(counter < 500)
+        return { StartingSquare: { Row: "A", Column: -1 }, EndingSquare : { Row: "A", Column: -1 } };
+    }
+
+    private detectCollision(shipPlaces: ShipPlace[]): boolean { //apart from the last one, none of these for loops should be more than 5 iterations so it's not as bad as it looks
+        for (let i: number = 0; i < shipPlaces.length - 1; i++) {
+            let ship1: Position[] = this.generateShipSquares(shipPlaces[i]);
+            for (let j: number = i + 1; j < shipPlaces.length; j++) {
+                let ship2: Position[] = this.generateAdjacentSquares(this.generateShipSquares(shipPlaces[j]));
+                for (let x: number = 0; x < ship1.length; x++) {
+                    for (let y: number = 0; y < ship2.length; y++) {
+                        if ((ship1[x].Row === ship2[y].Row) && (ship1[x].Column === ship2[y].Column)) {
+                            return true;
+                        }
+                    }
+                }
+            }
+        }
+        return false;
+    }
+
+    private generateShipSquares(shipPlace: ShipPlace): Position[] {
+        let ship: Position[] = [];
+        if (shipPlace.StartingSquare.Row === shipPlace.EndingSquare.Row) {
+            for (let i: number = shipPlace.StartingSquare.Column; i <= shipPlace.EndingSquare.Column; i++) {
+                ship.push({
+                    Row: shipPlace.StartingSquare.Row,
+                    Column: i
+                });
+            }
+        } else {
+            for (let i: number = shipPlace.StartingSquare.Row.charCodeAt(0); i <= shipPlace.EndingSquare.Row.charCodeAt(0); i++) {
+                ship.push({
+                    Row: String.fromCharCode(i),
+                    Column: shipPlace.StartingSquare.Column
+                });
+            }
+        }
+        return ship;
+    }
+
+    private generateAdjacentSquares(ship: Position[]): Position[] {
+        if (ship === []) {
+            return []
+        }
+        let squares: Position[] = [];
+        for (let i: number = 0; i < ship.length; i++) {
+            squares.push({ Row: ship[i].Row, Column: ship[i].Column + 1 });
+            squares.push({ Row: ship[i].Row, Column: ship[i].Column - 1 });
+            squares.push({ Row: String.fromCharCode(ship[i].Row.charCodeAt(0) + 1), Column: ship[i].Column });
+            squares.push({ Row: String.fromCharCode(ship[i].Row.charCodeAt(0) - 1), Column: ship[i].Column });
+            squares.push({ Row: String.fromCharCode(ship[i].Row.charCodeAt(0) + 1), Column: ship[i].Column + 1 });
+            squares.push({ Row: String.fromCharCode(ship[i].Row.charCodeAt(0) - 1), Column: ship[i].Column + 1 });
+            squares.push({ Row: String.fromCharCode(ship[i].Row.charCodeAt(0) + 1), Column: ship[i].Column - 1 });
+            squares.push({ Row: String.fromCharCode(ship[i].Row.charCodeAt(0) - 1), Column: ship[i].Column - 1 });

```

Comment:
This would be much nicer if this had a proper type - e.g.

`squares.push (new RowColumn(ship[i].Row, ship[i].Column + 1));`